libraries/core_libs/network/src/tarcap/packets_handlers/get_pbft_sync_packet_handler.cpp

```diff
@@ -32,38 +32,58 @@ void GetPbftSyncPacketHandler::process(const PacketData &packet_data,
   // Here need PBFT chain size, not synced period since synced blocks has not verified yet.
   const size_t my_chain_size = pbft_chain_->getPbftChainSize();
   size_t blocks_to_transfer = 0;
+  auto will_be_synced = false;
   if (my_chain_size >= height_to_sync) {
-    blocks_to_transfer = std::min(network_sync_level_size_, (my_chain_size - (height_to_sync - 1)));
+    auto total_sync_blocks_size = my_chain_size - height_to_sync + 1;
+    if (total_sync_blocks_size <= network_sync_level_size_) {
+      blocks_to_transfer = total_sync_blocks_size;
+      will_be_synced = true;
+    } else {
+      blocks_to_transfer = network_sync_level_size_;
+    }
+    LOG(log_tr_) << "Will send " << blocks_to_transfer << " PBFT blocks to " << packet_data.from_node_id_;
+  } else {
+    LOG(log_wr_) << "Request syncing period is ahead of own PBFT chain. Peer " << packet_data.from_node_id_
+                 << " want to sync PBFT blocks period from " << height_to_sync << ". But own PBFT chain size is "
+                 << my_chain_size;
+    will_be_synced = true;
   }
-  LOG(log_tr_) << "Will send " << blocks_to_transfer << " PBFT blocks to " << packet_data.from_node_id_;
 
-  sendPbftBlocks(packet_data.from_node_id_, height_to_sync, blocks_to_transfer);
+  sendPbftBlocks(packet_data.from_node_id_, height_to_sync, blocks_to_transfer, will_be_synced);
 }
 
 // api for pbft syncing
 void GetPbftSyncPacketHandler::sendPbftBlocks(dev::p2p::NodeID const &peer_id, size_t height_to_sync,
-                                              size_t blocks_to_transfer) {
+                                              size_t blocks_to_transfer, bool will_be_synced) {
   LOG(log_tr_) << "sendPbftBlocks: peer want to sync from pbft chain height " << height_to_sync
                << ", will send at most " << blocks_to_transfer << " pbft blocks to " << peer_id;
-  uint64_t current_period = height_to_sync;
+  dev::RLPStream s;
+  auto pbft_chain_synced = false;
   if (blocks_to_transfer == 0) {
```

Comment:
In case there are no blocks to be sent, we should probably never even get into this function... See my other comments for better understanding

Reply:
I agree. FIxed, thanks